### PR TITLE
leading distribution docs

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
https://github.com/flutter/flutter/issues/72521

- TODO: interactive example with sliders after getting this to work on web

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/painting/text_style.dart [hunk 1]
```diff
@@ -150,6 +150,47 @@ const double _kDefaultFontSize = 14.0;
...
+///  * Configuration 2: same as Configuration 1, except [TextHeightBehavior.applyHeightToFirstAscent] is set to false.
+///  * Configuration 3: [leadingDistribution] is set to [TextLeadingDistribution.even].
+///  * Configuration 4: same as Configuration 3, except [TextHeightBehavior.applyHeightToLastDescent] is set to false.
+///
+/// The [leadingDistribution] property controls how leading is distributed over
+/// and under the text. With [TextLeadingDistribution.proportional]
+/// (Configuration 1), `Top Leading : Bottom Leading = Font Ascent : Font
+/// Descent`, which also means the alphabetic baseline divides the line height
+/// into 2 parts proportional to the font's ascent and descent. With
+/// [TextLeadingDistribution.proportional] (Configuration 3), `Top Leading`
```
Comment (justinmc) [inline packages/flutter/lib/src/painting/text_style.dart:182]: Typo, this should be [TextLeadingDistribution.even]

**Review by justinmc — COMMENTED** [review]
One typo that needs to be fixed, otherwise LGTM.

Thank you for improving these docs!  I think this should be great on the docs site with your diagram.

+1 to the interactive example when it's possible.

**Review by justinmc — APPROVED** [review]
LGTM 👍   Thanks for the typo fix.

**Review by Oklahomawhore — COMMENTED** [review]
funky

**Pull request merged**